pkg/scheduler/framework/runtime/framework.go

```diff
@@ -655,7 +656,16 @@ func (f *frameworkImpl) RunPreFilterPlugins(ctx context.Context, state *framewor
 		if !s.IsSuccess() {
 			s.SetFailedPlugin(pl.Name())
 			if s.IsUnschedulable() {
-				return nil, s
+				if s.Code() == framework.UnschedulableAndUnresolvable {
```

Comment:
Comparing to always continue running PreFilter upon `Unschedulable`, I'm thinking of _conditional_ continue/stop: 

- option 1: offer a knob in scheduler config to indicate whether it's intended to continue run PreFilter plugin upon the first `Unschedulable` - it should be a profile-specific parameter.
- option 2: try to deduce the user's intent w/o offering a config knob. Technically, it's doable to tell whether a PreFilter plugin returns nil in PreFilterExtensions. For example, suppose pluginA and pluginB both implement PreFilter, pluginB is placed after pluginA. When pluginA fails with `Unschedulable`
  - runPreFilterPlugins() stops if pluginB returns nil PreFilterExtensions
  - runPreFilterPlugins() continues if pluginB returns non-nil PreFilterExtensions

Reply:
sg. I prefer option2, as option1 seems a bit difficult for users to understand appropriately, and I don't come up with any scenario that option2 cannot cover, but people want to continue or stop running other PreFilter.

---

But, can we leave this enhancement to a follow-up? because I'm thinking this PR deserves to be cherry-picked to older versions.

Reply:
cc @alculquicondor @ahg-g @kerthcet A simple approach is to keep running all PreFilter plugins upon Unschedulable, which should be no penalty for in-tree plugins (at least for now).

Reply:
btw: even for out-of-tree plugins, it's still very strict to trigger this issue, basically you have to satisfy the following conditions:

https://github.com/kubernetes/kubernetes/issues/119770#issuecomment-1666664335

> - an out-of-tree pluginA returns Unschedulable in PreFilter
> - an out-of-tree pluginB is placed after pluginA, and
>   - it implements PreFilter, and write its own key to cycleState
>   - it implements PreFilterExtensions#Add/RemovePod, and report error upon missing key

Reply:
Sorry, I haven't got time to check the sig-meeting rec, but is the conclusion just to go with keeping running all PreFilter when Unschedulable and leave the idea of conditional continue at least for now?
Also, is there any blocker on this PR that I need to pile up to get `/lgtm`?

Reply:
I'm fine with the current approach. And it's not that severe comparing to other 2 regressions we recently discussed. Let's polish this (I guess there will be conflicts on tests) after those 2 regression PRs get merged first, and then discuss if we want to cherry-pick.